frontend/src/utils/address.ts

```diff
@@ -43,20 +44,9 @@ export const lookupEnsName = async (address: string, provider: Provider) => {
 export const lookupCnsName = async (address: string) => {
   try {
     // Send request to get names
-    const url = 'https://api.thegraph.com/subgraphs/name/unstoppable-domains-integrations/dot-crypto-registry';
-    const res = await fetch(url, {
-      method: 'POST',
-      headers: { 'Content-Type': 'application/json' },
-      body: JSON.stringify({
-        variables: { owner: address.toLowerCase() },
-        query: 'query domainsOfOwner($owner: String!) { domains(where: {owner: $owner}) { name } }',
-      }),
-    });
-
-    // Return the first name in the array, or null if user has no CNS names
-    const json = (await res.json()) as CnsQueryResponse;
-    const names = json.data.domains;
-    return names.length > 0 ? names[0].name : null;
+    const resolution = new Resolution();
+    const domain = await resolution.reverse(address);
```

Comment:
Can you explain how forward and resolution should work across chains? For example, here's what we do for ENS:
- If using mainnet, forward and reverse resolution is done using mainnet contracts. Reverse resolution (from address to name) is validated by ensuring the forward lookup for the name matches.
- If rinkeby, same as mainnet, but using rinkeby contracts.
- For all other networks, use the mainnet resolution flow. If withdrawing to an ENS name, and it resolves to a contract, warn that funds may not be accessible on the current network

Reply:
Yeah, the implementation is same as ENS